packages/flutter/lib/src/widgets/form.dart

```diff
@@ -137,13 +154,17 @@ class Form extends StatefulWidget {
 /// Typically obtained via [Form.of].
 class FormState extends State<Form> {
   int _generation = 0;
+  bool _hasInteractedByUser = false;
   final Set<FormFieldState<dynamic>> _fields = <FormFieldState<dynamic>>{};
 
   // Called when a form field has changed. This will cause all form fields
   // to rebuild, useful if form fields have interdependencies.
   void _fieldDidChange() {
     if (widget.onChanged != null)
       widget.onChanged();
+    for (final FormFieldState<dynamic> field in _fields) {
+      _hasInteractedByUser = field._hasInteractedByUser == true;
```

Comment:
This seems wrong, only the last one will affect the result. Also you should be able to exit early if one of them is true, you can do _fields.any(..)